File: Fragment_Communication-II-03Aug21/app/src/main/java/com/example/fragmentcommunication_iii03aug21/StudentPersonalDetailsFragment.java
package com.example.fragmentcommunication_iii03aug21;

import android.content.Context;
import android.os.Bundle;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.Button;
import android.widget.EditText;

public class StudentPersonalDetailsFragment extends Fragment {
    private Button mBtnNext;
    private EditText mEtName, mEtAge;
    private CommunicationListener communicationListener;
    @Override
    public View onCreateView(LayoutInflater inflater, ViewGroup container,
                             Bundle savedInstanceState) {
        return inflater.inflate(R.layout.fragment_student_personal_details, container, false);

    }

    @Override
    public void onAttach(@NonNull Context context) {
        super.onAttach(context);
        this.communicationListener = (CommunicationListener) context;
    }

    @Override
    public void onViewCreated(@NonNull View view, @Nullable Bundle savedInstanceState) {
        super.onViewCreated(view, savedInstanceState);
        initViews(view);
    }

    private void initViews(View view) {
        mBtnNext = view.findViewById(R.id.btnNext);
        mEtAge = view.findViewById(R.id.etStudentAge);
        mEtName = view.findViewById(R.id.etStudentName);

        mBtnNext.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                String name = mEtName.getText().toString();
                int age = Integer.parseInt(mEtAge.getText().toString());
                Bundle bundle = new Bundle();
                bundle.putString("name", name);
                bundle.putInt("age", age);
                if(communicationListener != null) {
                    communicationListener.LaunchPerformanceFragment(bundle);
                }
            }
        });
    }
}

File: _Kotlin Projects/Navigation_Graph_IW_20Aug21/app/src/main/java/com/example/navigation_graph_iw_20aug21/StudentPersonalDetailsFragment.kt
package com.example.navigation_graph_iw_20aug21

import android.os.Bundle
import androidx.fragment.app.Fragment
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.navigation.NavController
import androidx.navigation.Navigation
import kotlinx.android.synthetic.main.fragment_student_personal_details.*

class StudentPersonalDetailsFragment : Fragment(R.layout.fragment_student_personal_details) {

    private lateinit var navController: NavController

    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
        super.onViewCreated(view, savedInstanceState)

        navController = Navigation.findNavController(view)

        btnNext.setOnClickListener {
            val name = etStudentName.text.toString()
            val age = etStudentAge.text.toString().toInt()
            var bundle = Bundle()
            bundle.putString("name", name)
            bundle.putInt("age", age)
            navController.navigate(
                R.id.action_studentPersonalDetailsFragment_to_studentPerformanceDetailsFragment,
                bundle
            )
        }
    }
}